tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__Handler.java

```diff
@@ -61,20 +61,23 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             // TODO: handle command
 
             // Note: if communication with thing fails for some reason,
-            // indicate that by setting the status with detail information
+            // indicate that by setting the status with detail information:
             // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
             // "Could not control device at IP address x.x.x.x");
         }
     }
 
     @Override
     public void initialize() {
+        // logger.debug("Start initializing!");
         config = getConfigAs(${bindingIdCamelCase}Configuration.class);
 
         // TODO: Initialize the thing. If done set status to ONLINE to indicate proper working.
-        // Long running initialization should be done asynchronously in background.
+        // Long running initialization (e.g. WAN access) should be done asynchronously in the background. Set a status 
+        // of UNKNOWN or OFFLINE in these situations and return quickly. Set the status to ONLINE once a proper 
```

Comment:
I think it does not become clear here that if you do async init, the status when returning from this method should be set to UNKNOWN (do you see any valid usecase for OFFLINE as well?) and the async code must then change it to ONLINE or OFFLINE asap.
Especially line 81 below can probably easily confuse people what to do here.

Reply:
I agree and admit that I had a hard time even for this small sentence ;-)
I will rephrase the whole paragraph.

Reply:
done.